cannot make install

URGENT
Cannot install everything properly because every time I try to do ``make install``, whether in 22.04 Ubuntu or 24.04 Ubuntu, and regardless of what CMake I myself use, the CMake requirements for dcm2niix are not allowed by the system. See the output below to ``make install``:
```bash



-- Installing Shimming Toolbox -- 
See our website for more information: https://shimming-toolbox.org
 
Clean install, deleting /home/cmhales/shimming-toolbox. Warning: If you have previously installed fsleyes-plugin-shimming-toolbox, this will delete it.


...
...


Installing fsleyes-plugin-shimming-toolbox
...

Looking for: ['python=3.10']

conda-forge/linux-64                                        Using cache
conda-forge/noarch                                          Using cache
Transaction

  Prefix: /home/cmhales/shimming-toolbox/python

  All requested packages already installed


Installing shimming-toolbox

...
...

  Building wheel for dcm2niix (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for dcm2niix (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [127 lines of output]
      /tmp/pip-build-env-8_h48mrt/overlay/lib/python3.10/site-packages/setuptools_scm/_integration/setuptools.py:92: UserWarning: version of None already set
        warnings.warn(f"version of {dist_name} already set")
      
      
      --------------------------------------------------------------------------------
      -- Trying 'Ninja' generator
      --------------------------------
      ---------------------------
      ----------------------
      -----------------
      ------------
      -------
      --
      CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
        Compatibility with CMake < 3.10 will be removed from a future version of
        CMake.
      
        Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
        to tell CMake that the project requires at least <min> but has been updated
        to work with policies introduced by <max> or earlier.
      
      Not searching for unused variables given on the command line.
      
      -- The CXX compiler identification is GNU 13.3.0
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: /usr/bin/g++ - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      -- Configuring done (0.1s)
      -- Generating done (0.0s)
      -- Build files have been written to: /tmp/pip-install-ixwfmzkh/dcm2niix_d52d9f6482f744a5b406c2729df9858e/_cmake_test_compile/build
      --
      -------
      ------------
      -----------------
      ----------------------
      ---------------------------
      --------------------------------
      -- Trying 'Ninja' generator - success
      --------------------------------------------------------------------------------
      
      Configuring Project
        Working directory:
          /tmp/pip-install-ixwfmzkh/dcm2niix_d52d9f6482f744a5b406c2729df9858e/_skbuild/linux-x86_64-3.10/cmake-build
        Command:
          /tmp/pip-build-env-8_h48mrt/overlay/lib/python3.10/site-packages/cmake/data/bin/cmake /tmp/pip-install-ixwfmzkh/dcm2niix_d52d9f6482f744a5b406c2729df9858e -G Ninja -DCMAKE_MAKE_PROGRAM:FILEPATH=ninja --no-warn-unused-cli -DCMAKE_INSTALL_PREFIX:PATH=/tmp/pip-install-ixwfmzkh/dcm2niix_d52d9f6482f744a5b406c2729df9858e/_skbuild/linux-x86_64-3.10/cmake-install -DPYTHON_VERSION_STRING:STRING=3.10.17 -DSKBUILD:INTERNAL=TRUE -DCMAKE_MODULE_PATH:PATH=/tmp/pip-build-env-8_h48mrt/overlay/lib/python3.10/site-packages/skbuild/resources/cmake -DPYTHON_EXECUTABLE:PATH=/home/cmhales/shimming-toolbox/python/bin/python -DPYTHON_INCLUDE_DIR:PATH=/home/cmhales/shimming-toolbox/python/include/python3.10 -DPYTHON_LIBRARY:PATH=/home/cmhales/shimming-toolbox/python/lib/libpython3.10.so -DPython_EXECUTABLE:PATH=/home/cmhales/shimming-toolbox/python/bin/python -DPython_ROOT_DIR:PATH=/home/cmhales/shimming-toolbox/python -DPython_FIND_REGISTRY:STRING=NEVER -DPython_INCLUDE_DIR:PATH=/home/cmhales/shimming-toolbox/python/include/python3.10 -DPython3_EXECUTABLE:PATH=/home/cmhales/shimming-toolbox/python/bin/python -DPython3_ROOT_DIR:PATH=/home/cmhales/shimming-toolbox/python -DPython3_FIND_REGISTRY:STRING=NEVER -DPython3_INCLUDE_DIR:PATH=/home/cmhales/shimming-toolbox/python/include/python3.10 -DCMAKE_MAKE_PROGRAM:FILEPATH=ninja -DCMAKE_BUILD_TYPE:STRING=Release
      
      CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
        Compatibility with CMake < 3.10 will be removed from a future version of
        CMake.
      
        Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
        to tell CMake that the project requires at least <min> but has been updated
        to work with policies introduced by <max> or earlier.
      
      Not searching for unused variables given on the command line.
      
      -- The C compiler identification is GNU 13.3.0
      -- The CXX compiler identification is GNU 13.3.0
      -- Detecting C compiler ABI info
      -- Detecting C compiler ABI info - done
      -- Check for working C compiler: /usr/bin/gcc - skipped
      -- Detecting C compile features
      -- Detecting C compile features - done
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: /usr/bin/g++ - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      -- Found Git: /usr/bin/git (found version "2.43.0")
      -- Configuring done (0.3s)
      -- Generating done (0.0s)
      -- Build files have been written to: /tmp/pip-install-ixwfmzkh/dcm2niix_d52d9f6482f744a5b406c2729df9858e/_skbuild/linux-x86_64-3.10/cmake-build
      [1/9] Creating directories for 'console'
      [2/9] No download step for 'console'
      [3/9] No update step for 'console'
      [4/9] No patch step for 'console'
      [5/9] Performing configure step for 'console'
      FAILED: console-prefix/src/console-stamp/console-configure /tmp/pip-install-ixwfmzkh/dcm2niix_d52d9f6482f744a5b406c2729df9858e/_skbuild/linux-x86_64-3.10/cmake-build/console-prefix/src/console-stamp/console-configure
      cd /tmp/pip-install-ixwfmzkh/dcm2niix_d52d9f6482f744a5b406c2729df9858e/_skbuild/linux-x86_64-3.10/cmake-build/console-build && /tmp/pip-build-env-8_h48mrt/overlay/lib/python3.10/site-packages/cmake/data/bin/cmake -Wno-dev --no-warn-unused-cli -DCMAKE_INSTALL_PREFIX:PATH=/tmp/pip-install-ixwfmzkh/dcm2niix_d52d9f6482f744a5b406c2729df9858e/_skbuild/linux-x86_64-3.10/cmake-build -DCMAKE_C_COMPILER:FILEPATH=/usr/bin/gcc -DCMAKE_C_FLAGS:STRING= -DCMAKE_CXX_COMPILER:FILEPATH=/usr/bin/g++ -DCMAKE_CXX_FLAGS:STRING= -DCMAKE_VERBOSE_MAKEFILE:BOOL=FALSE -DUSE_STATIC_RUNTIME:BOOL=ON -DUSE_TURBOJPEG:BOOL=OFF -DUSE_JASPER:BOOL=OFF -DUSE_JPEGLS:BOOL=OFF -DUSE_JNIFTI:BOOL=ON -DZLIB_IMPLEMENTATION:STRING=Miniz -DZLIB_ROOT:PATH= -DUSE_OPENJPEG:BOOL=OFF -DOpenJPEG_DIR:PATH= -DBATCH_VERSION:BOOL=OFF -DYAML-CPP_DIR:PATH= -DBUILD_DCM2NIIXFSLIB:BOOL=OFF -GNinja -S /tmp/pip-install-ixwfmzkh/dcm2niix_d52d9f6482f744a5b406c2729df9858e/console -B /tmp/pip-install-ixwfmzkh/dcm2niix_d52d9f6482f744a5b406c2729df9858e/_skbuild/linux-x86_64-3.10/cmake-build/console-build && /tmp/pip-build-env-8_h48mrt/overlay/lib/python3.10/site-packages/cmake/data/bin/cmake -E touch /tmp/pip-install-ixwfmzkh/dcm2niix_d52d9f6482f744a5b406c2729df9858e/_skbuild/linux-x86_64-3.10/cmake-build/console-prefix/src/console-stamp/console-configure
      Not searching for unused variables given on the command line.
      -- The C compiler identification is GNU 13.3.0
      -- The CXX compiler identification is GNU 13.3.0
      -- Detecting C compiler ABI info
      -- Detecting C compiler ABI info - done
      -- Check for working C compiler: /usr/bin/gcc - skipped
      -- Detecting C compile features
      -- Detecting C compile features - done
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: /usr/bin/g++ - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      CMake Error at ucm.cmake:13 (cmake_minimum_required):
        Compatibility with CMake < 3.5 has been removed from CMake.
      
        Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
        to tell CMake that the project requires at least <min> but has been updated
        to work with policies introduced by <max> or earlier.
      
        Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
      Call Stack (most recent call first):
        CMakeLists.txt:6 (include)
      
      
      -- Configuring incomplete, errors occurred!
      ninja: build stopped: subcommand failed.
      Traceback (most recent call last):
        File "/tmp/pip-build-env-8_h48mrt/overlay/lib/python3.10/site-packages/skbuild/setuptools_wrap.py", line 668, in setup
          cmkr.make(make_args, install_target=cmake_install_target, env=env)
        File "/tmp/pip-build-env-8_h48mrt/overlay/lib/python3.10/site-packages/skbuild/cmaker.py", line 696, in make
          self.make_impl(clargs=clargs, config=config, source_dir=source_dir, install_target=install_target, env=env)
        File "/tmp/pip-build-env-8_h48mrt/overlay/lib/python3.10/site-packages/skbuild/cmaker.py", line 741, in make_impl
          raise SKBuildError(msg)
      
      An error occurred while building with CMake.
        Command:
          /tmp/pip-build-env-8_h48mrt/overlay/lib/python3.10/site-packages/cmake/data/bin/cmake --build . --target install --config Release --
        Install target:
          install
        Source directory:
          /tmp/pip-install-ixwfmzkh/dcm2niix_d52d9f6482f744a5b406c2729df9858e
        Working directory:
          /tmp/pip-install-ixwfmzkh/dcm2niix_d52d9f6482f744a5b406c2729df9858e/_skbuild/linux-x86_64-3.10/cmake-build
      Please check the install target is valid and see CMake's output for more information.
      
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for dcm2niix
Successfully built shimmingtoolbox
Failed to build dcm2niix
ERROR: Could not build wheels for dcm2niix, which is required to install pyproject.toml-based projects
make: *** [Makefile:17: install] Error 1


```

--- a/installer/install_shimming_toolbox.sh
+++ b/installer/install_shimming_toolbox.sh
@@ -40,7 +40,7 @@ function edit_shellrc() {
 source "${ST_DIR}/${PYTHON_DIR}/bin/activate"
 
 print info "Installing python"
-"$ST_DIR"/"${PYTHON_DIR}"/bin/mamba install -y -c conda-forge python=3.10 dcm2niix=1.0.20241211
+"$ST_DIR"/"${PYTHON_DIR}"/bin/mamba install -y -c conda-forge python=3.10
 
 print info "Installing shimming-toolbox"
 cp "${ST_PACKAGE_DIR}/shimmingtoolbox/config/dcm2bids.json" "${ST_DIR}/dcm2bids.json"

--- a/shimming-toolbox/requirements_st.txt
+++ b/shimming-toolbox/requirements_st.txt
@@ -11,7 +11,7 @@ numpy>=1.21
 phantominator~=0.6.4
 pillow>=9.0.0
 psutil>=5.8.0
-pydicom
+pydicom~=2.4.4
 pytest>=6.2.5
 pytest-cov>=2.5.1
 quadprog

--- a/shimming-toolbox/requirements_st.txt
+++ b/shimming-toolbox/requirements_st.txt
@@ -2,6 +2,7 @@ click
 cloup
 dataclasses
 dcm2bids>=3.0.1
+dcm2niix>=1.0.20250506
 importlib-metadata
 joblib
 matplotlib>=3.5

